src/UniswapV2Callee.sol

```diff
@@ -107,13 +138,36 @@ contract UniswapV2CalleeDai is UniswapV2Callee {
         uint256 daiToJoin = divup(daiAmt, RAY);
 
         // Do operation and get dai amount bought (checking the profit is achieved)
-        uniRouter02.swapExactTokensForTokens(
-            gemAmt,
-            add(daiToJoin, minProfit),
-            path,
-            address(this),
-            block.timestamp
-        );
+        try gem.token0() returns (TokenLike tokenA) { // gem is an LP token
+            TokenLike tokenB = gem.token1();
+            uniRouter02.removeLiquidity({ // burn token to obtain its components
+                tokenA: address(tokenA),
+                tokenB: address(tokenB),
+                liquidity: gemAmt,
+                amountAMin: 0, // minProfit is checked below
+                amountBMin: 0,
```

Comment:
I'm trying to figure out if deferring the check will cause problems.  Basically, it will cost more gas for a failed transaction if we put the check off.  In the worst case, I'm thinking of a sandwich attacking bot blindly trying to sandwich based on there being no slippage restriction, and then when this fails, our transaction costing more.

Reply:
Can you give more detail regarding the possible attack? I'm not super familiar with sandwich attacks.

Also, have you thought of a possible solution? Because the user-provided `minProfit` variable only makes sense if it is tested at the end, when we know exactly how much profit the trade had overall.

AFAICT, the best we can do is add some slippage guardrail instead of zero. For this, we would need access to price data. Would that be an acceptable solution to this sandwich problem?

Reply:
I think this is okay because of the minProfit constraint. It constrains the whole operation to be profitable including any possible front-running at any step.

Reply:
I guess I really just don't know what bots are looking for when they sandwich.  But if they do it blindly based on a `0` output amount and not checking the constraint at the end passes, then everyone looses and this code would cost more.

I'm okay with deploying this and seeing if such bots exist, and then if they do we just have to fix this and redeploy.

Reply:
Well it doesn't really matter what they do because any manipulation beyond the minProfit will result in tx failure. My guess is the bots will evaluate the whole tx block including their sandwich/frontrun/etc to see if it's worthwhile. With the minProfit specified it should not be (or only minimally).

Reply:
@godsflaw your concern seems a bit prophetic now